app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,15 @@
+class ShoppingListCntroller < ApplicationController
+  def create
+    @shopping_list = ShoppingList.create(shopping_list_params)
+    @shopping_list.items.create(name: 'Pinapple')
+    @shopping_list.items.create(name: 'Apple')
+    @shopping_list.items.create(name: 'Chicken breasts')
+    redirect_to @shopping_list
+  end
+
+  private
+
+  def shopping_list_params
+    params.require(:shopping_list).permit(:name)
+  end
```

Comment:
Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).

you can put the logic to check between recipes ingredients and foods then calculate the missing food